core/shared/utils/bh_vector.c

```diff
@@ -65,6 +81,22 @@ bh_vector_init(Vector *vector, size_t init_length, size_t size_elem)
     vector->size_elem = size_elem;
     vector->max_elems = init_length;
     vector->num_elems = 0;
+    vector->lock = NULL;
+
+    if (use_lock) {
+        if (!(vector->lock = wasm_runtime_malloc(sizeof(korp_mutex)))) {
+            LOG_ERROR("Init vector failed: alloc locker failed.\n");
+            bh_vector_destroy(vector);
+            return false;
+        }
+
+        if (BHT_ERROR == os_mutex_init(vector->lock)) {
```

Comment:
Should use `0 != os_mutex_init` or `BHT_OK != os_mutex_init`, as os_mutex_init is implemented by various platforms, the return value may be not BHT_ERROR when we failed to init it.

Reply:
got it